docs/docs/how-tos/How to install and configure Keycloak.md

```diff
@@ -0,0 +1,155 @@
+
+    :::note
+    Although not required, users may not be able to log in to Grafana if this field isn't properly set.
+    :::
+
+   - Associate the user with one or more of the **Groups**. Out of the box, Nebari is deployed with the following groups: ``admin``, ``analyst``, and ``developer`` (see the [Roles x Groups](#roles-x-groups) section below for more details).
+
+4. Click **Save**.
+
+
+If you are using the password authentication provider, you will also need to define a password for the user. Keep in mind that, if using Auth0 or Github the password field is not required. It's best to put the **Temporary** toggle in the **OFF** position. Otherwise the user will be forced to change the password on first login.
+
+
+  ![Keycloak add user > credentials tab screenshot - set password](/img/how-tos/keycloak_user_password.png)
+
+
+### Add user from the command line
+
+To make adding users easier for new Nebari deployments, the follwoing client command will help you with this.
+
+```shell
+nebari keycloak -c nebari-config.yaml adduser <username> <password>
+```
+
+This will create a new user `<username>` under the ``analyst`` group, with the initial password provided. Omit the password completely if you are using GitHub or Auth0.
+
+:::note
+If you [changed the initial root password for Keycloak](#change-keycloak-root-password) this method will not work.
+:::
+
+## Login to Nebari
+
+Your new user can now log in to Nebari, visit your provided Nebari domain URI which will take you to the login form page and follow the [How to login in to Nebari and start a server](/how-tos/login-keycloak).
+
+![Nebari - Log in to Keycloak page](/img/how-tos/nebari_login_screen.png)
+
+## Roles x Groups
+
+Groups represent a collection of users that perform similar actions and therefore require similar permissions. By default, Nebari is deployed with the following groups: ``admin``,
+``developer``, ``analyst`` and ``viewer`` (in roughly descending order of power).
+
+| Group       | Access to Nebari Resources                                                                    | Roles | Permissions Description |
+| ----------- | ------------------------------------------------------------------------------------------- | ----- | -------------------- |
+| ``admin``     | <ul><li>Conda-Store</li><li>Dask</li><li>Jupyterhub</li><li>Grafana</li></ul> |   <ul><li>``conda_store_admin``</li><li>``dask_admin``</li><li>``jupyterhub_admin``</li><li>``grafana_admin``</li></ul> | Blank |
+| ``developer`` | <ul><li>Conda-Store</li><li>Dask</li><li>Jupyterhub</li><li>Grafana Developer</li></ul> | <ul><li>``conda_store_developer``</li><li>``dask_developer``</li><li>``jupyterhub_developer``</li><li>``grafana_developer``</li></ul> | Blank |
+| ``analyst``   | <ul><li>Conda-Store</li><li>Jupyterhub</li><li>Grafana</li></ul>  | <ul><li>``conda_store_developer``</li><li>``jupyterhub_developer``</li><li>``grafana_viewer``</li></ul> | Blank |
```

Comment:
@costrouc I tried to expand this a litlle bit, to give more info regarding this permission level system, would be happy if you could add some thoughts regarding the missing description for this 'roles'... e.g. what precisely a user with this permission will be able to do, and what not  (for example, the difference between grafana admin and viewer, dask, etc.)